clang/include/clang/StaticAnalyzer/Core/PathSensitive/BasicValueFactory.h

```diff
@@ -165,79 +166,70 @@ class BasicValueFactory {
 
   /// Convert - Create a new persistent APSInt with the same value as 'From'
   ///  but with the bitwidth and signedness of 'To'.
-  const llvm::APSInt &Convert(const llvm::APSInt& To,
-                              const llvm::APSInt& From) {
+  APSIntPtr Convert(const llvm::APSInt &To, const llvm::APSInt &From) {
     APSIntType TargetType(To);
     if (TargetType == APSIntType(From))
-      return From;
+      return APSIntPtr(&From);
```

Comment:
This might create a tricky bug, given that despite the doc comment and the type signature, this function does not create a persistent APSInt *only in this case*, if `From` was not persistent on the call.
I suggest requiring the `From` argument to be `APSIntPtr` to make this function safer.

Reply:
Makes sense. I'm opting for the safe API, and I'll just call `getValue()`.